tolstoy/src/syncer.rs

```diff
@@ -155,20 +155,70 @@ impl Syncer {
             // On succes:
             // - persist local mappings from the receiver
             // - update our local "remote head".
-            TxMapper::set_bulk(&mut db_tx, &uploader.tx_temp_uuids)?;
-            SyncMetadataClient::set_remote_head(&db_tx, &last_tx_uploaded)?;
+            TxMapper::set_bulk(db_tx, &uploader.tx_temp_uuids)?;
+            SyncMetadataClient::set_remote_head(db_tx, &last_tx_uploaded)?;
+        }
+
+        Ok(())
+    }
+
+    pub fn flow(sqlite: &mut rusqlite::Connection, server_uri: &String, user_uuid: &Uuid) -> Result<()> {
+        d(&format!("sync flowing"));
 
-            // Commit everything: tx->uuid mappings and the new HEAD. We're synced!
-            db_tx.commit()?;
+        ensure_current_version(sqlite)?;
```

Comment:
Future: `e_c_v` expects to be called during open or create, so it establishes its own transaction. You can probably  omit it here, but if you don't you should figure out how to share the transaction.